### PR TITLE
docbook package restructuring

### PR DESCRIPTION
**Pull request by JPeterMugaas**
docbook-xml - remove 5.x series (that's a total rewrite)
docbook5-xml - new package that includes docbook 5.0, 5.0.1, and 5.1
w3c-mathml2 - new package MathML XML scheme
opengl-man-pages - new package - opengl documentaiton (uses docbook5-xml, docbook 4.x) and we

Since opengl-man-pages builds, I'm assume that this works as it does use some of the old packaging stuff and I ddidn't see an error when testing xmlcat with the -v parameter.

### CODE REVIEW

File: mingw-w64-docbook5-xml/docbook5-xml-i686.install [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  if [ -e ${MINGW_XML_CATALOG}/catalog.preserve ]; then
+    mv ${MINGW_XML_CATALOG}/catalog.preserve ${MINGW_XML_CATALOG}/catalog
+  elif [ ! -e ${MINGW_INSTALL}/etc/xml/catalog ]; then
+    ${MINGW_INSTALL}/bin/xmlcatalog --noout --create ${MINGW_XML_CATALOG}/catalog
+  fi
+
+  for v in 5.{0,0.1,1}; do
+      ${MINGW_INSTALL}/bin/xmlcatalog --noout --add "delegatePublic" \
+        "-//OASIS//DTD DocBook XML ${v}//EN" \
+        "file://${_datadir}/xml/docbook/schema/dtd/${v}/catalog.xml" \
```
Comment (Alexpux) [inline mingw-w64-docbook5-xml/docbook5-xml-i686.install]: Where you get ${_datadir}?

File: mingw-w64-docbook5-xml/docbook5-xml-x86_64.install [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  if [ -e ${MINGW_XML_CATALOG}/catalog.preserve ]; then
+    mv ${MINGW_XML_CATALOG}/catalog.preserve ${MINGW_XML_CATALOG}/catalog
+  elif [ ! -e ${MINGW_INSTALL}/etc/xml/catalog ]; then
+    ${MINGW_INSTALL}/bin/xmlcatalog --noout --create ${MINGW_XML_CATALOG}/catalog
+  fi
+
+  for v in 5.{0,0.1,1}; do
+      ${MINGW_INSTALL}/bin/xmlcatalog --noout --add "delegatePublic" \
+        "-//OASIS//DTD DocBook XML ${v}//EN" \
+        "file://${_datadir}/xml/docbook/schema/dtd/${v}/catalog.xml" \
```
Comment (Alexpux) [inline mingw-w64-docbook5-xml/docbook5-xml-x86_64.install]: _datadir?

**Comment by Alexpux** [comment]
@JPeterMugaas you kidding me? 
Look how our packages for docbook done with relative paths and do the same.

File: mingw-w64-docbook5-xml/docbook5-xml-x86_64.install (hunk 2 above)
Comment (JPeterMugaas) [reply mingw-w64-docbook5-xml/docbook5-xml-x86_64.install]: Alright, I did attempt to fix this with cygpath -m as well as fix the catalog entries.  Hopefully, this works.

Comment (JPeterMugaas) [reply mingw-w64-docbook5-xml/docbook5-xml-x86_64.install]: Sorry for any delays.  I wanted to be sure that I was doing things properly.  Docbook does rely upon the xmlcatalog and sometimes, that's unclear.

File: mingw-w64-w3c-mathml2/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Contributor: Sylvain HENRY <hsyl20@gmail.com>
+
+_realname=w3c-mathml2
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=2.0
+pkgrel=5
```
Comment (Alexpux) [inline mingw-w64-w3c-mathml2/PKGBUILD]: Set pkgrel to 1

File: mingw-w64-w3c-mathml2/w3c-mathml2-i686.install [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+_rootcatalog=$(cygpath -m ${MINGW_INSTALL}/etc/xml/catalog)
+
+post_install() {
+  if [ ! -e etc/xml/catalog ]; then
+    ${MINGW_INSTALL}/bin/xmlcatalog --noout --create ${MINGW_INSTALL}/etc/xml/catalog
+  fi
+
+  ${MINGW_INSTALL}/bin/xmlcatalog  --noout --add public "-//W3C//DTD MathML 2.0//EN" "${_rootcatalog}/share/xml/w3c/mathml2/mathml2.dtd" ${MINGW_INSTALL}/eetc/xml/catalog
+  ${MINGW_INSTALL}/bin/xmlcatalog  --noout --add public "-//W3C//DTD MathML//EN" "${_rootcatalog}/share/xml/w3c/mathml2/mathml2.dtd" ${MINGW_INSTALL}/eetc/xml/catalog
+  ${MINGW_INSTALL}/bin/xmlcatalog  --noout --add system "http://www.w3.org/TR/MathML2/dtd/mathml2.dtd" "${_rootcatalog}/share/xml/w3c/mathml2/mathml2.dtd" ${MINGW_INSTALL}/eetc/xml/catalog
```
Comment (Alexpux) [inline mingw-w64-w3c-mathml2/w3c-mathml2-i686.install]: Here is typo in 3 lines ````eetc````.

Also you need do as in previous packages with relative paths

File: mingw-w64-docbook5-xml/docbook5-xml-i686.install [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  if [ -e ${MINGW_XML_CATALOG}/catalog.preserve ]; then
+    mv ${MINGW_XML_CATALOG}/catalog.preserve ${MINGW_XML_CATALOG}/catalog
+  elif [ ! -e ${MINGW_INSTALL}/etc/xml/catalog ]; then
+    ${MINGW_PREFIX}/bin/xmlcatalog --noout --create ${MINGW_XML_CATALOG}/catalog
+  fi
+
+  for v in 5.{0,0.1,1}; do
+      ${MINGW_PREFIX}/bin/xmlcatalog --noout --add "delegatePublic" \
+        "-//OASIS//DTD DocBook XML ${v}//EN" \
+        "${_subcatalogdir}/docbook-${v}.xml" \
```
Comment (Alexpux) [inline mingw-w64-docbook5-xml/docbook5-xml-i686.install]: @JPeterMugaas I tell you many times that **YOU MUST NOT USE ABSOLUTE PATHS HERE!!!!!!** See current docbook-xml package

**Comment by Alexpux** [comment]
@JPeterMugaas packages is still broken. Do not use absolute paths as you do in post install scripts. Look at current docnbook-xml package and do similar. Also package w3c-mathml2 have the same issues - you must not add toxml catalog records with absolute paths

**Comment by Alexpux** [comment]
@JPeterMugaas now it looks more relevant. Will see it myself now on my PC

**Pull request merged**